funsor/minipyro.py

```diff
@@ -60,7 +61,7 @@ def expand_inputs(self, name, size):
 #     See http://docs.pyro.ai/en/0.3.1/parameters.html
 
 PYRO_STACK = []
-PARAM_STORE = {}
+PARAM_STORE = ParamStoreDict()  # A dict-like object that also supports constraints.
```

Comment:
I guess this is OK for now, but the `ParamStoreDict` code is pretty gross.  It really only needs to be a dictionary whose values are tuples `(unconstrained_value, constraint)`, with `pyro.param` responsible for applying the constraint.  Maybe we should fix that in Pyro's minipyro and then port the fix to funsor.